Scrum 23 fe page create post

MariaZasypkina opened the pull request:
Added form fields, category dropdown, image upload preview
Matched overall layout and styling
Included Post and Cancel buttons with proper navigation

src/util/categories.js  is created to use until categories / subcategories list is finalized  

Added <Route path="/dev-postcreate" element={<PostCreate />} />  for development and testing outside of PrivateRoute protection

Note: theme styles are not implemented fully and to be added in compliance with other branches components 

<img width="601" alt="Screenshot 2025-04-10 at 12 08 19 AM" src="https://github.com/user-attachments/assets/acb6bcec-118e-404d-8e7e-5782b8b8d10a" />

Cherrymood commented on src/components/Router.jsx:
```diff
@@ -44,6 +44,8 @@ export default function AppRouter() {
                 <Route path="profile" element={<Profile />} />
               </Route>
 
+              <Route path="/dev-postcreate" element={<PostCreate />} />
```
more often <Route path="/new" element={<PostCreate />} />

AidaBur replied:
I think this route was added just to check the PostCreate page without going through PrivateRoute.
Just a small reminder to maybe remove or comment it out so it doesn’t stay in the router by mistake.

Cherrymood commented on src/pages/PostCreate.jsx:
All functions are already in POSTSContext API, you should
//Check the adding PostsContext.jsx - uncomment posts
1.import { usePosts } from "../context/PostsContext";
2. const { createPost} = usePosts(); 
3. Post new data in function createPost(newData);

MariaZasypkina replied:
thank you!  Fixed!

Cherrymood commented on src/util/categories.js:
everything is in dummy file

MariaZasypkina replied:
Do you mean that all categories are in dummyItems for now and you suggest to take them directly from there?  I decided since that’s just temporary, I it makes sense to keep them in a separate categories.js file.
I thought it might be easier to update later when we connect the backend — we could just replace the source without touching the component.
If that’s not the best way to do it, I can change it — just not sure it’s really necessary right now

Cherrymood reviewed (commented):
I really like you job! And I have wrote some things to improve.

AidaBur commented on src/pages/PostCreate.jsx:
```diff
@@ -1,4 +1,140 @@
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    const newPost = {
+      item_id: Date.now(),
```
Since createPost in the context already generates the item_id, maybe we don't need to duplicate that logic here in the component?

MariaZasypkina replied:
thank you Aida! fixed it

AidaBur commented:
All design mockups are available in the shared Figma file (previously shared).
Let's try to keep a consistent visual style across all pages as we build them.

MariaZasypkina commented:
Hey everyone, 

here are updates made to  the  PostCreate to match design and review feedback

- Removed item_id generation from component; now handled in PostsContext
- Replaced local setPosts with context's createPost function
- Applied Tailwind theme variables (e.g., focus:ring-primary) instead of hardcoded styles
- Improved image upload UX with preview and conditional rendering
- Finalized layout and text styles to match prototype

MariaZasypkina commented on src/pages/PostCreate.jsx:
```diff
@@ -1,4 +1,140 @@
+          <input
+            name="title"
+            placeholder="Title"
+            className="w-full px-4 py-3 border border-gray-300 rounded-xl text-sm focus:outline-none focus:ring-2 focus:ring-[#76D900]" // TODO: use focus:ring-primary
```
You're right, I just had issues with tailwind and used hardcoded as a temporary step, now everything should be ok

Cherrymood commented:
Great! looks good!

pavel6767 commented on src/pages/PostCreate.jsx:
```diff
@@ -1,4 +1,173 @@
+  };
+
+  // Handle image upload for local preview
+  const handleImageChange = (e) => {
```
since the photo change is not handled by the `handleChange` function, I'd separate the states, use one state for text and boolean inputs and have another useState for photos, especially since we're targetting to allow the user to upload many photos

something like
```javascript
const [photos, setPhotos] = useState([])

when adding a photo
... newPhoto = { url, URL.create..., photo: photo}
setPhotos( [...photos, newPhoto]

also I'd add a handler to remove a photo that the user doesn't want to upload
```

pavel6767 commented on tailwind.config.js:
```diff
@@ -5,8 +5,17 @@ export default {
     ],
     theme: {
-      extend: {},
+      extend: {
```
nice, theme variables

mateuspiresl reviewed (approved):
The submit button is transparent here. If it's a problem on my end only, don't worry about it.

![Screenshot 2025-04-11 at 20 27 27](https://github.com/user-attachments/assets/f0ab5529-67a6-4dc8-b385-4e5044d425ae)

MariaZasypkina commented:
I resolved a merge conflict and re-requested approval reviews
